### DISCUSSION TITLE
Google Login giving me a un usual error.

### DISCUSSION THREAD

**Discussion by jlwegaba** [datamweb/shield-oauth#41] [Q&A]
Hi @datamweb ,

All was well when on my localhost. Could get user registered and logged into my system from GOOGLE perfectly well.

However, moving onto my CPANEL Host server (its shared), got a very weird 404 error "**Sorry! Cannot seem to find the page you were looking for.**" and on activating back to development environment, the error said "**no route get: 403.shtml"**

And also weird bit, no error is registered in error_log.

Have tried but all in vain:-
- By passing sheild-oauth to do one directly either as php file or another controller method.
- renaming the call back link
- Installed the gogole-services-for-php onto the server, still nothing.

Any help will be appreciated please.

NOTE: Other Sheild-oauth are working fine, that is Yahoo, Facebook

**Comment by datamweb** [comment]
Hey @jlwegaba, sorry for the delay, I'm traveling.

I tried to test online on shared hosting, I didn't see any problem.
The link below is the online demo.

https://codeigniter4.ir/oauth/google

So I can't figure out where the problem is, have you made any special changes? Please provide what version(ci4, shield and shield-oauth) you are using and **anything you think I can do to reproduce the problem**.

Also, please provide a **screenshot in development mode** of the error you received.

**Reply by jlwegaba** [reply]
Hello @datamweb 
Thanks alot for the response.

Here is the link https://bakla.ug/oauth/google 

so am able to get to get to first screen of choosing gmail to use (see screenshot)
![oauth1](https://github.com/datamweb/shield-oauth/assets/99025174/85fd7036-2af8-498a-b1ec-2d45c7442a9a)

As soon as i choose an account to authenticate with below happens.
![image](https://github.com/datamweb/shield-oauth/assets/99025174/0f038a12-074b-407f-a846-41f3b5c8d2c5)

Here is browser response.
https://bakla.ug/oauth/call-back?state=027b69d1839a205491783e0e496cf2a276073635&code=4%2F0AZEOvhWvLC0zZwXutBsVbAqkpHuk9qlPiMR7GuREJ8eQkO_jEy-OCVPkJIGdkhnCOCqJoA&scope=email+profile+openid+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.email+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.profile&authuser=0&prompt=consent

NOTE:
- I only made one change i asked sometime back of "**social_provider**" as another field to know where user is logging with.
- My installation is not through composer since i kept getting issues of not able to install through it so i used the manual one.

I have for now activated the login with google https://bakla.ug/login. I will keep it active for next hours

**Reply by jlwegaba** [reply]
This is my callbak link: https://bakla.ug/oauth/call-back

**Reply by datamweb** [reply]
@jlwegaba I tried hard to reproduce your problem, but unfortunately I couldn't. You may have made some changes that I am not aware of.
Can you upload a complete mini project (ci4, shield and shield-oauth) that produces your error as a zip file?

**Reply by datamweb** [reply]
I will check the attached file, it may take some time.
> but can share direct access to server. please let me know if i can email you details

There is no need at this stage. I will inform you if needed.

**Reply by datamweb** [reply]
Hi @jlwegaba file you sent did not help me.
The file below is a ready project with minimal code.
Make the changes you made so I can see your problem.

Submit the file after you have successfully reproduced the problem.

This project is ready, just follow the instructions below.

```console
php spark migrate --all
```

```console
php spark serve
```

[appstarter.zip](https://github.com/datamweb/shield-oauth/files/12448244/appstarter.zip)

**Reply by jlwegaba** [reply]
Hey @datamweb Thank you very much for your efforts.

I solved the issue when i moved the website to another server. It was i think server configs that were messing me up because I tested the same setting on 2 other sites on server server and produced same problem and when moved to another worked immediately.

All is good now.

Thank you very much for the useful oauth plugin.
